component/settings/HNNFlexLayoutContainer.js

```diff
@@ -0,0 +1,550 @@
+		this.state = {
+			modelExist: false,
+			canvasUpdateRequired: false,
+			simulationUpdateRequired: true,
+			dipoleHTML: null,
+			tracesHTML: null,
+			psdHTML: null,
+			rasterHTML: null,
+			spectrogramHTML: null,
+			hnnInstantiatedVisible: true,
+			dipoleIframeVisible: true,
+			tracesIframeVisible: false,
+			psdIframeVisible: false,
+			rasterIframeVisible: false,
+			spectrogramIframeVisible: false,
+		};
+	}
+
+	async componentDidMount(prevProps, prevState) {
+		const { dipoleHTML } = this.state;
+
+		if (dipoleHTML===null) {
+			const message = 'hnn_geppetto.get_dipole_plot';
+
+			Utils.evalPythonMessage(message,[]).then(response => {
+				let html_quoted = response.replace(/\\n/g, '').replace(/\\/g, '');
+				let html = html_quoted.substring(1, html_quoted.length-1);
+				this.setState({ dipoleHTML: html });
+			  })
+		}
+	}
+
+	async componentDidUpdate(prevProps, prevState) {
+		const { showCanvas } = this.props;
+		const { modelExist } = this.state;
+		// when showing the canvas, check if the model has changed
+		// to know if we need to re-run simulation or update the canvas
+		if (showCanvas && !prevProps.showCanvas && modelExist) {
+			const message = 'hnn_geppetto.compare_cfg_to_last_snapshot';
+			const { canvasUpdateRequired, simulationUpdateRequired } = await Utils.evalPythonMessage(message, []);
+			this.setState({ canvasUpdateRequired, simulationUpdateRequired });
+		}
+
+		if((this.state.dipoleIframeVisible !== prevState.dipoleIframeVisible) && this.state.dipoleIframeVisible) {
+			this.addTabToTabSetOrCreate("Top",{
+				"name": "Dipole",
+				"component": "DipoleIframe",
+				"id": "dipole"
+			});
+		}
+		if((this.state.tracesIframeVisible !== prevState.tracesIframeVisible) && this.state.tracesIframeVisible) {
+			this.addTabToTabSetOrCreate("Bottom",{
+				"name": "Traces",
+				"component": "TracesIframe",
+				"id": "traces"
+			});
+		}
+		if((this.state.psdIframeVisible !== prevState.psdIframeVisible) && this.state.psdIframeVisible) {
+			this.addTabToTabSetOrCreate("Bottom",{
+				"name": "PSD",
+				"component": "PSDIframe",
+				"id": "psd"
+			});
+		}
+		if((this.state.rasterIframeVisible !== prevState.rasterIframeVisible) && this.state.rasterIframeVisible) {
+			this.addTabToTabSetOrCreate("Bottom",{
+				"name": "Raster",
+				"component": "RasterIframe",
+				"id": "raster"
+			});
+		}
+		if((this.state.spectrogramIframeVisible !== prevState.spectrogramIframeVisible) && this.state.spectrogramIframeVisible) {
+			this.addTabToTabSetOrCreate("Bottom",{
+				"name": "Spectrogram",
+				"component": "SpectrogramIframe",
+				"id":"spectrogram"
+			});
+		}
+
+		if((this.state.hnnInstantiatedVisible !== prevState.hnnInstantiatedVisible) && this.state.hnnInstantiatedVisible) {
+			this.addTabToTabSetOrCreate("Bottom", {
+				"name": "3D",
+				"component": "HNNInstantiated",
+				"id": "3d"
+			});
+			this.instantiate()
+		}
+	
+	}
+
+	addTabToTabSetOrCreate(location, json){
```

Comment:
Hey @ddelpiano you mind to take a look at this and let me know if you find something obviously incorrect? 
The purpose of this function is:
- add dipole plots to a new tabset 
- add a new tab to the bottom tabset if it exists in any other case.
- create a new tabset in the bottom if there is none.

Reply:
It seems ok, you are using a logic slightly different from what I did but I suppose this works (never tried) since the root node should RowNode and we cannot remove that and you check that condition in the first if, so either way you add your element in that node. In my case I checked if a tabset was present, if not that is one scenario where I add the tabset and then I add the node to this tabset.
If this works for you it's even better than the solution I found since I am using the flexlayout's internal API to achieve what I want to do and this seems to be cleaner.
Let me know when you have a live version since I want to test manually few scenarios and in case I will steal this logic :-).

Reply:
It seems ok, you are using a logic slightly different from what I did but I suppose this works (never tried) since the root node should RowNode and we cannot remove that and you check that condition in the first if, so either way you add your element in that node. In my case I checked if a tabset was present, if not that is one scenario where I add the tabset and then I add the node to this tabset.
If this works for you it's even better than the solution I found since I am using the flexlayout's internal API to achieve what I want to do and this seems to be cleaner.
Let me know when you have a live version since I want to test manually few scenarios and in case I will steal this logic :-).